tests/unit/server/test_utils.py

```diff
@@ -37,6 +37,7 @@
         ({"max_keys": 1}, ["key1:value1,value2,value3"], {"key": "key1", "value": ["value1", "value2", "value3"]}),
         ({"max_keys": 1, "max_values_per_key": 1}, ["key1:value1"], {"key": "key1", "value": "value1"}),
         ({"max_values_per_key": 1}, ["key1:value1", "key2:value2"], {"key1": "value1", "key2": "value2"}),
+        ({"max_values_per_key": 1}, ["key1", "key2"], {"key1": None, "key2": None}),
```

Comment:
should we also set the `group_keys_without_values=False` to achieve the expected result?